### PR TITLE
Fix MC Trajectory plotting in TEve

### PR DESCRIPTION
**Pull request by sophiemiddleton**
- was missing an "abs" so only plotting some charges

### CODE REVIEW

**Comment by kutschke** [comment]
@sophie A gentle reminder that this PR is waiting for a followup from you.

**Comment by sophiemiddleton** [comment]
What is the follow up?

**Comment by kutschke** [comment]
> What is the follow up?

There are 3 comments in the conversation.  Two ask for changes.

**Comment by sophiemiddleton** [comment]
I dont see any comments - on the code or in the conversation. Can you repost?

**Comment by kutschke** [comment]
> I dont see any comments - on the code or in the conversation. Can you repost?

Are you logged in to GitHub?   I only see the comments when I am logged in.

**Comment by kutschke** [comment]
Do you have time  now for a zoom session?


Topic: Rob Kutschke's Personal Meeting Room



Join Zoom Meeting

https://fnal.zoom.us/j/6308405645?pwd=ZU00azR6WTR4b0FLb0pBMkJMdE9Pdz09



Meeting ID: 630 840 5645

Passcode: mu2e


From: Sophie Middleton ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Thursday, March 17, 2022 at 11:21 AM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Fix MC Trajectory plotting in TEve (PR #730)


Yes, I'm logged in, but I still dont see anyting . The PR is just changing two lines of code

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_730-23issuecomment-2D1071041682&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=eJ3b2Pwd9QQ-ELi_SvfEkwTMLbjamkNQFc-bllJbaaIZ-eNw9n8o9HVa0jCQsiPM&s=W7QHcAdFx05Uqe30WlcmwJf2UThTjezMNE6ff5_HDWo&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZSMCPPIHCSJHPSEUSTVANLWZANCNFSM5QHYL6CQ&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=eJ3b2Pwd9QQ-ELi_SvfEkwTMLbjamkNQFc-bllJbaaIZ-eNw9n8o9HVa0jCQsiPM&s=6SB5KSumzvqmJmS9bQBAS5ahQsZ1baw1PAJgKRK9vf0&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=eJ3b2Pwd9QQ-ELi_SvfEkwTMLbjamkNQFc-bllJbaaIZ-eNw9n8o9HVa0jCQsiPM&s=5kPkE1tm7COzCqmzp5BhRks0mQrpW6zyP6YzHkE_c-4&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=eJ3b2Pwd9QQ-ELi_SvfEkwTMLbjamkNQFc-bllJbaaIZ-eNw9n8o9HVa0jCQsiPM&s=nDOzhkuE65aDGe4RDN1NEbOibPH-PdobHpnra-9B1xU&e=>.

**Comment by kutschke** [comment]
The reason that Sophie could not see my comments but I could is that I had not hit the final submit button - so they were pending.

File: TEveEventDisplay/src/TEveMu2eMCInterface.cc [hunk 1]
```diff
@@ -115,7 +115,7 @@ namespace mu2e{
               posy = 1500.0;
               break;
           case PDGCode::mu_minus:
-              color = kGreen;
+              color = kOrange-7;
```
[the remark this answers is not in the document]
Comment (sophiemiddleton) [reply TEveEventDisplay/src/TEveMu2eMCInterface.cc:118]: kGreen is used for the KalSeed tracks so I wanted to use a different color. The -7 was to make it more distringuishavle from other colors

**Comment by sophiemiddleton** [comment]
Comments have been addressed. As for multiple particles with same code, I guess you would draw all of those, I dont think its an issue, currently they would all be "other" anyway.

File: TEveEventDisplay/src/TEveMu2eMCInterface.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kutschke) [reply TEveEventDisplay/src/TEveMu2eMCInterface.cc:118]: Please add a comment that the 7 is a tweak that happens to work, not something deeper than that.

**Comment by HCasler** [comment]
Looking into why this hasn't returned a result. Jenkins is reporting that the tests are all taking too long and getting killed, but I don't currently know why it's not reporting the failure.

**Comment by kutschke** [comment]
Sophie: please do NOT merge the main branch into your working branch unless you must resolve a conflict.  It's OK for now but in the future please do not do it unless it's necessary

**Comment by kutschke** [comment]
> Looking into why this hasn't returned a result. Jenkins is reporting that the tests are all taking too long and getting killed, but I don't currently know why it's not reporting the failure.

Helenka:   I noticed that there was a commit made after I launched the CI.  I suspect that this is not relevant but I want it on your radar just in case.

**Comment by HCasler** [comment]
@kutschke I don't think that made a difference; looking through the logs the actual merge was at commit [1f832ab.](https://github.com/Mu2e/Offline/commit/1f832ab59814bf9bbc498c9eb5c33c931c6bbeaf).

Individual test logs all indicate a failure with exit code 137. Also seem to indicate art barely even started, despite the processes getting killed for taking too long. It just shows the initial header with the Offline, KinKal, etc version, and then "Return Code: 137".

**Comment by sophiemiddleton** [comment]
@kutschke sorry - I assumed the reason for failure was some update I missed

**Comment by HCasler** [comment]
Update: still don't know why all the tests timed out. However, I do know why the build never reported a result. We have an overall timeout set in Jenkins. When we hit that time, the whole build fails immediately, but there's no action to report this failure. This PR hit that timeout sometime during muDauSteps.

Seems like when that happens, we need to report a timeout and set the label to "build failed".

**Comment by kutschke** [comment]
> @kutschke I don't think that made a difference; looking through the logs the actual merge was at commit [1f832ab.](https://github.com/Mu2e/Offline/commit/1f832ab59814bf9bbc498c9eb5c33c931c6bbeaf).
> 
> Individual test logs all indicate a failure with exit code 137. Also seem to indicate art barely even started, despite the processes getting killed for taking too long. It just shows the initial header with the Offline, KinKal, etc version, and then "Return Code: 137".

Thanks.  I will rerun and see what happens. Do you see any output from any of the jobs?  Do you understand why the information about failures was not posted to GitHub?

FYI unix return codes have the following convention: the code is one byte.  Values 0....127 are defined by the value returned from the main program.  Values greater than 128 mean that the program terminated with a unix signal (man signal to see the values).  So 137 is signal 9 - which just means that something killed the program.  That could come from a variety of environmental issues.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by rlcee** [comment]
I see the build also took about twice as long, 25 min:
Total build time: 1515.877450 second
I will check if this might be an effect of the Muse version change
though I don't know any mechanism.

**Comment by kutschke** [comment]
> Update: still don't know why all the tests timed out. However, I do know why the build never reported a result. We have an overall timeout set in Jenkins. When we hit that time, the whole build fails immediately, but there's no action to report this failure. This PR hit that timeout sometime during muDauSteps.
> 
> Seems like when that happens, we need to report a timeout and set the label to "build failed".

Good idea.  Please do this at your convenience.

**Comment by kutschke** [comment]
> I see the build also took about twice as long, 25 min:
> Total build time: 1515.877450 second
> I will check if this might be an effect of the Muse version change
> though I don't know any mechanism.

Thanks.  I also don't see a mechanism.

**Comment by kutschke** [comment]
Ok.  it worked.  We will put it down to gremlins.

**Pull request merged**